src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -613,138 +644,276 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
   /* Pin/Unpin
   /* ----------------------------------------------------------------------- */
 
+  currentCategoriesMatchLastPinnedCategories() {
+    // returns if the current categories match the last pinned categories
+    return this.lastPinnedCategory === this.category
+      && this.lastPinnedSubCategory === this.subCategory;
+  }
+
+  setLastPinnedCategories() {
+    this.lastPinnedCategory = this.category;
+    this.lastPinnedSubCategory = this.subCategory;
+  }
+
   getPinnedGenesCacheKey(category: string, subCategory?: string) {
     return (category + (subCategory ? '-' + subCategory : ''))
       .replace(/[^a-z0-9]/gi, '')
       .toLowerCase();
   }
 
-  getPinnedGenesCache(category: string, subCategory?: string) {
-    const key = this.getPinnedGenesCacheKey(category, subCategory);
-    return this.pinnedGenesCache[key] || [];
-  }
-
-  setPinnedGenesCache(genes: GCTGene[], category: string, subCategory: string) {
-    const key = this.getPinnedGenesCacheKey(category, subCategory);
-    this.pinnedGenesCache[key] = genes;
+  setPinnedGenesCache(genes: GCTGene[]) {
+    this.pinnedItemsCache = genes;
   }
 
-  clearPinnedGenesCache(category?: string, subCategory?: string) {
-    if (!category && !subCategory) {
-      this.pinnedGenesCache = {};
-    } else {
-      const key = this.getPinnedGenesCacheKey(category as string, subCategory);
-
-      if (subCategory) {
-        delete this.pinnedGenesCache[key];
-      } else {
-        for (const k in this.pinnedGenesCache) {
-          if (k.indexOf(key) === 0) {
-            delete this.pinnedGenesCache[k];
-          }
-        }
-      }
-    }
+  clearPinnedGenesCache() {
```

Comment:
And here as well -- `clearPinnedItemsCache()`

Reply:
done!